zerver/worker/queue_processors.py

```diff
@@ -609,3 +611,33 @@ def consume(self, event: Mapping[str, Any]) -> None:
                 (stream, recipient, sub) = access_stream_by_id(user_profile, stream_id,
                                                                require_active=False)
                 do_mark_stream_messages_as_read(user_profile, client, stream)
+        elif event['type'] == 'realm_exported':
+            realm = get_realm(event['realm_name'])
+            output_dir = tempfile.mkdtemp(prefix="zulip-export-")
+
+            to_s3 = False
+            if settings.LOCAL_UPLOADS_DIR is None:
+                to_s3 = True
```

Comment:
We should be passing `upload_to_s3` unconditionally; and then please do a follow-up commit to support uploading with the LOCAL_UPLOADS_DIR upload backend as well.

```
--- a/zerver/worker/queue_processors.py
+++ b/zerver/worker/queue_processors.py
@@ -612,25 +612,21 @@ class DeferredWorker(QueueProcessingWorker):
                                                                require_active=False)
                 do_mark_stream_messages_as_read(user_profile, client, stream)
         elif event['type'] == 'realm_exported':
-            realm = get_realm(event['realm_name'])
+            realm = Realm.objects.get(id=event['realm_id'])
             output_dir = tempfile.mkdtemp(prefix="zulip-export-")
 
-            to_s3 = False
-            if settings.LOCAL_UPLOADS_DIR is None:
-                to_s3 = True
+            # TODO: Add support for the LOCAL_UPLOADS_DIR uploads
+            # backend in export_realm_wrapper so we don't need this assertion.
+            assert settings.LOCAL_UPLOADS_DIR is None
 
-            export_realm_wrapper(realm=realm, output_dir=output_dir,
-                                 upload_to_s3=to_s3, threads=6, public_only=True,
-                                 delete_after_upload=to_s3)
+            public_url = export_realm_wrapper(realm=realm, output_dir=output_dir,
+                                              upload_to_s3=True, threads=6, public_only=True,
+                                              delete_after_upload=True)
+            assert public_url is not None
 
             admins = [user for user in realm.get_admin_users()]
-            content = "A public-only export was successfully completed!\n"
-            if to_s3:
-                content += (
-                    "The data was uploaded to the following url:\n\n"
-                    "http://%s.amazonaws.com/%s/%s" %
-                    (settings.S3_REGION, settings.S3_AUTH_UPLOADS_BUCKET, settings.S3_KEY,)
-                )
+            content = "Your data export is complete and has been uploaded here:\n\n%s" % (
+                public_url,)
 
             for user_profile in admins:
                 internal_send_private_message(
```